src/bannerClick.js

```diff
@@ -0,0 +1,6 @@
+import {promoObjFunc} from './modules/promoObj';
+
+let banner = document.getElementById('imageviewerOverlay');
+if (banner) {
+    banner.addEventListener('click', promoObjFunc('promotionClick', 'bts', 'Image viewer banner', 'Banner', 'On watermarked image'), true);
```

Comment:
Hi @punalsc - Is this right? Here you are passing a _call_ to `promoObjFunc()` as the second argument to `addEventListener`. I'm used to the second argument being a function reference. I think that passing it as a call will result in the function being called once (when the JavaScript first loads) and the returned object being provided as the handler to be invoked when the user clicks on the element. See here: 
https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener#The_event_listener_callback

Reply:
Good spotting. This is now fixed.